apis/clusters/v1beta1/kafka_types.go

```diff
@@ -459,3 +462,85 @@ func (rs *KafkaSpec) areDCsEqual(b []*KafkaDataCentre) bool {
 
 	return true
 }
+
+func (k *Kafka) NewExposePorts() []k8scorev1.ServicePort {
+	var ports []k8scorev1.ServicePort
+	ports = []k8scorev1.ServicePort{{
+		Name: models.SSH,
+		Port: models.Port22,
+		TargetPort: intstr.IntOrString{
+			Type:   intstr.Int,
+			IntVal: models.Port22,
+		},
+	},
+	}
+
+	if !k.Spec.PrivateNetworkCluster {
+		additionalPorts := []k8scorev1.ServicePort{
+			{
+				Name: models.KafkaClient,
+				Port: models.Port9092,
+				TargetPort: intstr.IntOrString{
+					Type:   intstr.Int,
+					IntVal: models.Port9092,
+				},
+			},
+			{
+				Name: models.KafkaControlPlane,
+				Port: models.Port9093,
+				TargetPort: intstr.IntOrString{
+					Type:   intstr.Int,
+					IntVal: models.Port9093,
+				},
+			},
+		}
+		if k.Spec.ClientToClusterEncryption {
+			sslPort := k8scorev1.ServicePort{
+				Name: models.KafkaBroker,
+				Port: models.Port9094,
+				TargetPort: intstr.IntOrString{
+					Type:   intstr.Int,
+					IntVal: models.Port9094,
+				},
+			}
+			additionalPorts = append(additionalPorts, sslPort)
+		}
+		ports = append(ports, additionalPorts...)
+	}
+
+	return ports
+}
+
+func (k *Kafka) NewHeadlessPorts() []k8scorev1.ServicePort {
+	ports := []k8scorev1.ServicePort{
+		{
+			Name: models.KafkaClient,
+			Port: models.Port9092,
+			TargetPort: intstr.IntOrString{
+				Type:   intstr.Int,
+				IntVal: models.Port9092,
+			},
+		},
+		{
+			Name: models.KafkaControlPlane,
+			Port: models.Port9093,
+			TargetPort: intstr.IntOrString{
+				Type:   intstr.Int,
+				IntVal: models.Port9093,
+			},
+		},
+	}
+	if k.Spec.ClientToClusterEncryption {
+		kafkaBrokerPort := k8scorev1.ServicePort{
+			Name: models.KafkaBroker,
+			Port: models.Port9094,
+			TargetPort: intstr.IntOrString{
+				Type:   intstr.Int,
+				IntVal: models.Port9094,
+			},
+		}
+		ports = append(ports, kafkaBrokerPort)
+	}
+
+	return ports
+}
```

Comment:
Can you move it to separate function and just path values you need to not duplicate such code a lot of times?

Reply:
We need to understand what kind of cluster we work with to decide what ports should be opened for it, so it will be difficult to make a pretty generic function